pinot-query-runtime/src/main/java/org/apache/pinot/query/service/SubmissionService.java

```diff
@@ -24,29 +24,30 @@
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
+import org.apache.pinot.common.exception.QueryException;
 
 
 /**
  * Submission service is used to submit multiple runnables and checks the result upon all {@link Future} returns
  * or any failure occurs.
  */
 public class SubmissionService {
-  private final ExecutorService _executor;
   private final List<CompletableFuture<Void>> _futures = new ArrayList<>();
 
-  public SubmissionService(ExecutorService executor) {
-    _executor = executor;
+  public SubmissionService() {
   }
 
-  public void submit(Runnable runnable) {
-    _futures.add(CompletableFuture.runAsync(runnable, _executor));
+  public void submit(Runnable runnable, ExecutorService executor) {
+    _futures.add(CompletableFuture.runAsync(runnable, executor));
   }
 
-  public void awaitFinish(long deadlineMs)
-      throws Exception {
+  public void awaitFinish(long deadlineMs) {
     CompletableFuture<Void> completableFuture = CompletableFuture.allOf(_futures.toArray(new CompletableFuture[]{}));
     try {
       completableFuture.get(deadlineMs - System.currentTimeMillis(), TimeUnit.MILLISECONDS);
+    } catch (Throwable t) {
```

Comment:
Can this ever throw `Throwable`? Or is it always wrapped in `ExecutionException`?

Reply:
it shouldn't be but it would be hell of a pain to debug if it did. so just to keep it safe. we should at least have a branch to handle it